### DISCUSSION TITLE
StoreController handler class cannot be loaded

### DISCUSSION THREAD

**Discussion by laceja** [phalcon/cphalcon#15946] [General]
Okay, I'm just trying to get started with Phalcon, taking the first tutorial to create an application called store.
Phalcon and the dev-tools are installed on ubuntu 20.04 with apache2 and php7.4
I type phalcon at the command prompt and get the basic help list as in the example. At the command prompt I'm in the root directory of /var/www/html, when I create the application with phalcon --create-project store.  I cd to the store folder and do an ls -al to be sure the application structure is there and it is.
Now, in my browser I type localhost/store and I get the following:
StoreController handler class cannot be loaded

#0 [internal function]: Phalcon\Mvc\Dispatcher->throwDispatchException('StoreController...', 2)
#1 [internal function]: Phalcon\Dispatcher\AbstractDispatcher->dispatch()
#2 /var/www/html/store/public/index.php(43): Phalcon\Mvc\Application->handle('/store/')
#3 {main}

What am I missing here. I've followed the very simple instructions to get this far. I'd really like to make use of phalcon and need someone to help me at least get started. I've searched the web and phalcon discussion groups and have found similar problems but, they are several years old and do not provide a viable solution. 
Please help!

**Comment by niden** [comment]
You will need to "tell" your application where your classes are. I don't recall the structure that the devtools creates but assume that you have something like this:

```
app/
    controllers/
    models/
    views/
```
and that your `StoreController` is located under `controllers`

You can either use the Phalcon Loader or composer - whichever is easier for you.

You will need to register the namespace and tell the loader where your files are. Assume that `MyProject` is your namespace:

_composer_
```
...
    "autoload": {
        "psr-4": {
            "MyProject\\Controllers\\": "app/controllers/",
            "MyProject\\Models\\": "app/models/"
        }
    },
...
```
or Phalcon Loader:
```php
<?php

use Phalcon\Loader;

$loader = new Loader();

$loader->registerNamespaces(
    [
       'MyProject\Controllers' => 'app/controllers',
       'MyProject\Models'      => 'app/models',
    ]
);

$loader->register();
```

For the Phalcon Loader, you can add that either in a `services.php` folder or your `index.php`

**Reply by laceja** [reply]
Thank you so much for your quick reply.
I'm sorry but, I'm still lost.  The code suggested for the phalcon loader already exists in the .htloader.php file and I have no idea how where to put the other code you suggested to use with composer.  I also tried adding the code to the index.php, which is located in the public folder with the same results.  I should also probably mention that my php experience is good, it's limited to the old version php5. I've no experience using namespaces, although I do understand the concept.  If there's anything else you can add that will help, it will be very much appreciated.
Merrill

**Reply by niden** [reply]
The .htaccess file is not used for autoloading. It merely directs traffic to one single script, in this case `public/index.php`. You have to tell your application where to find the clases you are referring to. 

Go to the https://github.com/phalcon/vokuro repository. You will notice the same or similar structure to what I described. Once the autoloader is set, you will just have to put your classes in the respective folders. 

Example:

https://github.com/phalcon/vokuro/blob/master/public/.htaccess#L7
Redirect all traffic to `index.html`- this is where the web server is pointing to)

https://github.com/phalcon/vokuro/blob/master/public/index.php#L18
index.php: Autoload stuff. This is the composer method I mentioned above. If you look at the `composer.json` file it loads the `Vokuro` namespace from the `src` folder. This means that if you have a namespaced class as `Vokuro/MyClass` it will try to find the file `src/MyClass.php`. It also requires a file as a helper.

https://github.com/phalcon/vokuro/blob/master/composer.json#L47

Now that the namespace is set and the application is instructed to load whatever it sees as `Vokuro/*` in the `src` folder, you can see here

https://github.com/phalcon/vokuro/tree/master/src

how things are structured. So in essence, if you need to load a controller your class will be:

`Vokuro/Controllers/AboutController` 

and the file will be 

`src/Controllers/AboutController.php`

A namespace is nothing special. It is a name that you use to group files together. With the autoloader you just tell your application where to load the files from.

**Comment by niden** [comment]
You can also have a look at Vokuro: https://github.com/phalcon/vokuro

It has a lot of Phalcon's functionality that you can utilize for your own project
